app/assets/javascripts/tictactoe.js

```diff
@@ -1 +1,203 @@
 // Code your JavaScript / jQuery solution here
+var turn;
+var gameId;
+$(document).ready(initialize)
+
+function initialize() {
+  attachListeners();
+  turn = 0;
+  gameId = undefined;
+}
+
+function player() { // X=even, Y=odd
+  if ((turn % 2) === 0) {
+    return "X"; 
+  } else {
+    return "O";
+  }
+}
+
+function updateState(el) { // Pass an element to update it with the current player's symbol
+  const x = el.attributes[0].nodeValue; // Get x coordinate (0-2)
+  const y = el.attributes[1].nodeValue; // Get y coordinate (0-2)
+  let square = $(`[data-x=${x}][data-y=${y}]`);
+
+  if (square.text() === "") {
+    square.text(player());
+    return true;
+  } else {
+    return false;
+  }
+}
+
+function setMessage(string) {
+  $("#message").text(string);
+}
+
+function checkWinner() {
+  let xSpots = [];
+  let o_spots = [];
+  let win = [
+    [0,1,2],[0,3,6],[0,4,8],[6,4,2],[2,5,8],[6,7,8],[3,4,5],[1,4,7]
```

Comment:
These winning combinations won't change, so it's best to declare them with `const`!